### PR TITLE
Add test coverage for Label 

### PR DESCRIPTION
**Pull request by Olina-Zhang**
Related https://github.com/dotnet/winforms/issues/10453

## Proposed changes

- Add unit tests for Label to test its properties: AutoEllipsis, BackgroundImage, BackgroundImageLayout, ImageIndex, ImageKey, ImageList, LiveSetting, UseCompatibleTextRendering
- Add unit tests for Label to test its events: AutoSizeChanged, BackgroundImageChanged, BackgroundImageLayoutChanged, KeyUp, KeyDown, KeyPress, TabStopChanged, TextAlignChanged

## Customer Impact

- None

## Regression? 

- No

## Risk

- Minimal

## Test methodology 

- Unit tests

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/dotnet/winforms/pull/10976)

### CODE REVIEW

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs [hunk 1]
```diff
@@ -421,6 +422,445 @@ public void Label_TextAlign_SetDifferentValue_Invalidate()
...
+        label.AutoSizeChanged -= handler;
+        label.AutoSize = !label.AutoSize;
+        Assert.Equal(2, callCount);
+    }
+
+    [WinFormsFact]
+    public void Label_AutoEllipsis_SetWithHandle_GetReturnsExpected()
+    {
+        using Label label = new();
+        Assert.NotEqual(0, label.Handle);
```
Comment (LeafShi1) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs:463]: Add a blank line under the Assert

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs [hunk 2]
```diff
@@ -421,6 +422,445 @@ public void Label_TextAlign_SetDifferentValue_Invalidate()
...
+        label.BackgroundImageLayoutChanged -= handler;
+        label.BackgroundImageLayout = ImageLayout.Zoom;
+        Assert.Equal(2, callCount);
+    }
+
+    [WinFormsFact]
+    public void Label_ImageIndex_GetSet_ImageList()
+    {
+        using ImageList imageList = new();
+        imageList.Images.Add(new Bitmap(10, 10));
```
Comment (Tanya-Solyanik) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs:613]: Please dispose of these bitmaps. This comment is applicable to other tests as well.

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs [hunk 3]
```diff
@@ -421,6 +422,454 @@ public void Label_TextAlign_SetDifferentValue_Invalidate()
...
+
+        label.BackgroundImageChanged += handler;
+        label.BackgroundImageChanged -= handler;
+        Assert.Equal(0, callCount);
+
+        label.BackgroundImageChanged += handler;
+        label.BackgroundImage = label.BackgroundImage;
+        Assert.Equal(0, callCount);
+
+        var image = new Bitmap(10, 10);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs:542]:
```suggestion
        using Bitmap image = new(10, 10);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs [hunk 4]
```diff
@@ -421,6 +422,454 @@ public void Label_TextAlign_SetDifferentValue_Invalidate()
...
+        label.BackgroundImageLayout = ImageLayout.Zoom;
+        Assert.Equal(2, callCount);
+    }
+
+    [WinFormsFact]
+    public void Label_ImageIndex_GetSet_ImageList()
+    {
+        using ImageList imageList = new();
+        using Bitmap bitmap1 = new Bitmap(10, 10);
+        using Bitmap bitmap2 = new Bitmap(10, 10);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs:615]: Please use target type new

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs [hunk 5]
```diff
@@ -421,6 +422,454 @@ public void Label_TextAlign_SetDifferentValue_Invalidate()
...
+    }
+
+    [WinFormsFact]
+    public void Label_BackgroundImageGetSet()
+    {
+        using Label label = new();
+        Assert.Null(label.BackgroundImage);  // Default value
+
+        // Set image.
+        using Bitmap image1 = new Bitmap(10, 10);
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs:504]: Please use target type new, ditto for some other tests below as well.
```suggestion
        using Bitmap image1 = new(10, 10);
```

File: src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs [hunk 6]
```diff
@@ -421,6 +422,454 @@ public void Label_TextAlign_SetDifferentValue_Invalidate()
...
+        using SubLabel label = new();
+        int callCount = 0;
+        KeyEventHandler handler = (sender, e) =>
+        {
+            Assert.Same(label, sender);
+            callCount++;
+        };
+
+        label.KeyUp += handler;
+        label.OnKeyUp(new KeyEventArgs(Keys.A));
```
Comment (lonitra) [inline src/System.Windows.Forms/tests/UnitTests/System/Windows/Forms/LabelTests.cs:750]: Consider making `new KeyEventArgs(Keys.A)` a variable and using it in both places we call `OnKeyUp`. Same suggestion for some other tests below

**Review by lonitra — COMMENTED** [review]
Some small suggestions, otherwise LGTM!

**Comment by Olina-Zhang** [comment]
Thank you all suggestion, avoid similar problems in the future as much as possible.

**Review by Tanya-Solyanik — APPROVED** [review]
Thank you!

**Review by lonitra — APPROVED** [review]

**Pull request merged**